test/automation/src/positron/positronConnections.ts

```diff
@@ -4,69 +4,60 @@
  *--------------------------------------------------------------------------------------------*/
 
 
+import { expect, Locator } from '@playwright/test';
 import { Code } from '../code';
 import { QuickAccess } from '../quickaccess';
-import { PositronBaseElement } from './positronBaseElement';
 
-const REMOVE_CONNECTION_BUTTON = 'a[aria-label="Remove connection from history"]';
-const DISCONNECT_BUTON = '.codicon-debug-disconnect';
-
-const PYTHON_CONNECTION_OPEN_STATE = 'div[aria-label="SQLite Connection"]';
-const R_CONNECTION_OPEN_STATE = 'div[aria-label="SQLiteConnection"]:first-child';
-const RECONNECT_BUTTON = 'a[aria-label="Execute connection code in the console"]';
-
-const CONNECTIONS_TAB_LINK = 'a[aria-label="Connections"]';
+const CONNECTIONS_CONTAINER = '.connections-items-container';
 
 /*
  *  Reuseable Positron connections tab functionality for tests to leverage
  */
 export class PositronConnections {
 
-	removeConnectionButton: PositronBaseElement;
-	disconnectButton: PositronBaseElement;
-	rConnectionOpenState: PositronBaseElement;
-	pythonConnectionOpenState: PositronBaseElement;
-	reconnectButton: PositronBaseElement;
-	connectionsTabLink: PositronBaseElement;
+	deleteConnectionButton: Locator;
+	disconnectButton: Locator;
+	connectIcon: Locator;
+	connectionItems: Locator;
 
 	constructor(private code: Code, private quickaccess: QuickAccess) {
-
-		this.removeConnectionButton = new PositronBaseElement(REMOVE_CONNECTION_BUTTON, this.code);
-		this.disconnectButton = new PositronBaseElement(DISCONNECT_BUTON, this.code);
-		this.rConnectionOpenState = new PositronBaseElement(R_CONNECTION_OPEN_STATE, this.code);
-		this.pythonConnectionOpenState = new PositronBaseElement(PYTHON_CONNECTION_OPEN_STATE, this.code);
-		this.reconnectButton = new PositronBaseElement(RECONNECT_BUTTON, this.code);
-		this.connectionsTabLink = new PositronBaseElement(CONNECTIONS_TAB_LINK, this.code);
+		this.deleteConnectionButton = code.driver.page.getByLabel('Delete Connection');
+		this.disconnectButton = code.driver.page.getByLabel('Disconnect');
+		this.connectIcon = code.driver.page.locator('.codicon-arrow-circle-right');
+		this.connectionItems = code.driver.page.locator('.connections-list-item');
```

Comment:
Nitpick, but I think Nick prefers just putting these outside the constructor since they are static.

Reply:
I _could_ do that, but they are dependent on `this.code.driver.page` since they are `Locators` and not just a string. which means I won't be able to use them like this in the test:
`await app.workbench.positronConnections.disconnectButton.click();`

Reply:
I thought locators were just static pointers to what would be used to find an element once they are used?

Reply:
Its really a nitpick... just keeping you in the loop regarding what Nick had mentioned perfering.